src/commands/synthetics/utils/internal.ts

```diff
@@ -253,11 +255,12 @@ export const validateAndParseOverrides = (overrides: string[] | undefined): Accu
           }
           break
 
-        // Convert to {[key: string]: string}
+        // Convert to StringObject
         case 'headers':
+        case 'variables':
           if (subKey) {
-            acc['headers'] = acc['headers'] ?? {}
-            acc['headers'][subKey] = value
+            const stringMap: StringMap = (acc[key] = acc[key] ?? {}) // acc[key] could be undefined, because of potential js Proxy
+            stringMap[subKey] = value
```

Comment:
The first line aligns better with the other cases, and the second line tells TS that `acc[key]` cannot truly be undefined in that case.

```suggestion
            acc[key] = acc[key] ?? {}
            ;(acc[key] as StringMap)[subKey] = value
```